client/data/settings/actions.js

```diff
@@ -0,0 +1,68 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { dispatch, select } from '@wordpress/data';
+import { apiFetch } from '@wordpress/data-controls';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import ACTION_TYPES from './action-types';
+import { NAMESPACE, STORE_NAME } from '../constants';
+
+function updateSettingsValues( payload ) {
+	return {
+		type: ACTION_TYPES.SET_SETTINGS_VALUES,
+		payload,
+	};
+}
+
+export function updateSettings( data ) {
+	return {
+		type: ACTION_TYPES.SET_SETTINGS,
+		data,
+	};
+}
+
+export function updateIsPaymentRequestEnabled( isEnabled ) {
+	return updateSettingsValues( { is_payment_request_enabled: isEnabled } );
+}
+
+export function updateIsSavingSettings( isSaving, error ) {
+	return {
+		type: ACTION_TYPES.SET_IS_SAVING_SETTINGS,
+		isSaving,
+		error,
+	};
+}
+
+export function* saveSettings() {
+	let error = null;
+	try {
+		const settings = select( STORE_NAME ).getSettings();
+
+		yield updateIsSavingSettings( true, null );
+
+		yield apiFetch( {
+			path: `${ NAMESPACE }/settings`,
+			method: 'post',
+			data: settings,
+		} );
+
+		yield dispatch( 'core/notices' ).createSuccessNotice(
+			__( 'Settings saved.', 'woocommerce-gateway-stripe' )
+		);
```

Comment:
(minor) Displaying a notice here caused a few quirks in WCPay. For example, we couldn't display a different messaging when the settings were saved (https://d.pr/i/2MGNwK) and it also displayed a notice when we didn't want to, on page navigation ( https://github.com/Automattic/woocommerce-payments/issues/2005 ).

What do you think about moving the notice logic somewhere else? Maybe in a separate component or via handlers on this function 🤷

Reply:
Extracted to a separate issue: #1840